lib/chef/provider/package/windows.rb

```diff
@@ -129,6 +129,7 @@ def action_install
         # Chef::Provider::Package action_install + action_remove call install_package + remove_package
         # Pass those calls to the correct sub-provider
         def install_package(name, version)
+          fail Chef::Exceptions::NoWindowsPackageSource, "installer_type '#{installer_type}' requires a source" if new_resource.source.nil?
```

Comment:
Not sure the `requirement_assert` will work here because the provider must be evaluated first which soes not occur until during the action. Just not sure how that ordering is setup but will play with it.